ipfs/provider.go

```diff
@@ -7,4 +7,4 @@ import (
 )
 
 // APIProvider allows customizable IPFS core APIs.
-type APIProvider func() (coreiface.CoreAPI, io.Closer, error)
+type APIProvider func() (coreiface.APIDagService, io.Closer, error)
```

Comment:
👍🏻 
This should make the IPFS resource usage footprint in tests very low!

Reply:
:crossed_fingers: